docs/src/components/hardware-list.js

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import hardware from "../data/hardware.json";
+
+const compatibleLookUp = {
+  pro_micro: "Pro Micro",
+  proton_c: "Proton-C",
+  makerdiary_m2: "makerdiary M.2",
```

Comment:
From @Nicell's original post:
> shields can also use a board's identifier in place of an interconnect if it's made for only one type of hardware
    this setting can also optionally be an array

I find the duality of `compatible` intriuging.

Interface specifications are either:
 a) defined from the outset (e.g. Adafruit Feather and nRF52840 M.2 Module)
 b) become "standard" based on the original board (e.g. [SparkFun] Pro Micro)

For (b), at what point does it switch from shields that fits a board (whose display name we presumably look up in the board's metadata), to shields that fits an interface "standard" defined in this lookup list.

Also, for the sake of argument, let's assume the Pro Micro ran ARM and thus ZMK.  Would the board and its namesake interface not compete for the same identifier: `pro_micro`.  We're lucky not to be in that situation now but it highlights the potential for conflicts in the duality.

It might be simpler to have something along the lines of:
```yaml
- compatibility
  - boards:
    -
  - interfaces:
    - 
```  
or just
```yaml
- boards:
  -
- interfaces:
  - 
```  

Thoughts anyone?

Reply:
For the pro micro case, it seems like them being the same is a *good* thing, since you don't need to retroactively change anything, when someone creates a second board that "mimics" the existing one.

Reply:
I can roll with that.  So the code would need to treat interfaces and boards on a priority basis ...
i.e. (psuedo code, without guards) `name = interfaces[id] ? interfaces[id].name : boards[id].name;`

Are there any fringe cases that would screw that approach up?

(I'm using the word interface rather than compatible for this discussion because it's less ambiguous)